### DISCUSSION TITLE
onDelete: CASCADE

### DISCUSSION THREAD

**Discussion by TheoMer** [prisma/prisma#3501] [Q&A] [answered]
Does Prisma 2 support this feature?

**Comment by ryands17** [comment, marked as answer]
In Prisma whenever you create relations, they are `onDelete: CASCADE` by default and currently it's only possible to configure it via SQL. There is an open request [here](https://github.com/prisma/prisma/issues/2810) and is also on our [roadmap](https://www.notion.so/prismaio/Prisma-public-roadmap-50766227b779464ab98899accb98295f?p=6a2f9b8550204992ba08ff7351c9afae).

**Reply by TheoMer** [reply]
Hey @ryands17 I have a 1 to 1 relation on two tables User and cartItem for which on delete cascade has been specified in postGreSQL. 

![foreign_key_rule](https://user-images.githubusercontent.com/15969262/91988579-568fe600-ed27-11ea-8580-8b2e1375b761.jpg)



But when I attempt to delete the User:

```
            await ctx.prisma.user.delete({
              where: { id: userId },
            }).catch(handleSubmitErr);
```


I get the following error message:

```
 Invalid `ctx.prisma.user.delete()` invocation in
C:\Users\d0475\Documents\Projects\flamingo-ecom2\sick-fits\backend\src\types\resolvers\Mutation.ts:634:41

   630 }
   631
   632 // 6. Delete guest_user
   633 if (userId && hasPermissions) {
→  634   await ctx.prisma.user.delete(
  The change you are trying to make would violate the required relation 'CartItemToUser' between the `CartItem` and `User` models.
```

My models are:

```
model CartItem {
  createdAt    DateTime      @default(now())
  id           String        @default(cuid()) @id
  item         String?
  itemvariants String?
  quantity     Int           @default(1)
  updatedAt    DateTime      @updatedAt
  user         String
  Item         Item?         @relation(fields: [item], references: [id])
  ItemVariants ItemVariants? @relation(fields: [itemvariants], references: [id])
  User         User          @relation(fields: [user], references: [id])
}

model User {
  createdAt        DateTime           @default(now())
  email            String             @unique
  id               String             @default(cuid()) @id
  name             String
  password         String
  permissions2     Permission2[]
  resetToken       String?
  resetTokenExpiry Float?
  updatedAt        DateTime           @updatedAt
  address          Address[]
  cart             CartItem[]
  items            Item[]
  itemvariants     ItemVariants[]
  order            Order[]
  orderitem        OrderItem[]
}
```

What's the issue here?

**Reply by ryands17** [reply]
@TheoMer this is an issue currently with Prisma that you can check here https://github.com/prisma/prisma/issues/2057

For this to work, your relations should be nullable i.e. a `?` added to the relation in your `schema.prisma` would work when deleting.

**Reply by TheoMer** [reply]
Hey @ryands17 that did the trick.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
